api/src/main/java/org/apache/iceberg/UpdateSchema.java

```diff
@@ -384,4 +384,25 @@ default UpdateSchema updateColumn(String name, Type.PrimitiveType newType, Strin
    *                                  with other additions, renames, or updates.
    */
   UpdateSchema unionByNameWith(Schema newSchema);
+
+  /**
+   * Add a new row identifier given the column name.
+   * <p>
+   * The column must be a required field of a primitive type.
+   * It must exist in the current schema to update, or is added as a part of this update.
+   *
+   * @param columnName name of the column to add as a row identifier
+   * @return this for method chaining
+   */
+  UpdateSchema addRowIdentifier(String columnName);
```

Comment:
Here also, I don't think that "row identifier" is the right noun. I'd prefer `addIdentifierField` and `removeIdentifierField`. I'd use `remove` rather than `delete` because it is more clear that the field itself will not be deleted, just removed from the identifier. I also think that we should make sure that behavior (removing from the identifier field list, not dropping the column) is clear in the Javadoc.

What about also adding `setIdentifierFields`? I think that a bulk set/replace operation fits more cleanly with the use case where users are setting the identifier for the first time or replacing it. I think that this will tend to be idempotent: set the identifier fields to `account_id` and `profile_id`.

Reply:
> I'd use remove rather than delete because it is more clear that the field itself will not be deleted

Sounds good, I also called it `remove` in the doc, `delete` was trying to match `deleteColumn`, but I agree it would cause some confusion.

> What about also adding setIdentifierFields

I thought about `setIdentifierFields`, the reason I did not add it was because the `UpdateSchema` interface seems to be all individual operations instead of bulk operations (except for `unionByNameWith`), and our DDL SQL statements are also not bulk updates. 

> I think that this will tend to be idempotent

I think both approaches are idempotent.  you can write it as:

```
update.setIdentifierFields(Sets.newHashSet("profile_id")).apply();
```

or 

```
update.addIdentifierField("profile_id").removeIdentifierField("account_id").apply();
```

I am fine with both approaches. Technically if we go with `setIdentifierFields`, then we do not need `addIdentifierField` and `removeIdentifierField` and I can try to figure out the diff for every update. 

@openinx do you think there is any advantage for any particular approach for the update API?

Reply:
Functionally,  there should be no difference between the `addIdentifierField`/`removeIdentifierField ` and `setIdentifierFields`,  but for the perspective of usability,  if people want to reset to use a totally different identifier fields  then he/she will need to: 

```java
    Schema schema = ...
    Set<Integer> newFieldIds = ...

    UpdateSchema update = new SchemaUpdate(SCHEMA, SCHEMA_LAST_COLUMN_ID);
    for (Integer identifier : schema.rowIdentifiers()) {
      String columnName = schema.findColumnName(identifier);
      Preconditions.checkNotNull(columnName, "Can not find the column with field id %s", identifier);

      update.deleteRowIdentifier(columnName);
    }


    for (Integer identifier : schema.rowIdentifiers()) {
      String columnName = schema.findColumnName(identifier);
      Preconditions.checkNotNull(columnName, "Can not find the column with field id %s", identifier);

      update.addRowIdentifier(columnName);
    }

    update.apply();
```

It does make it more difficult to use the API.  Besides the `setIdentifierFields`  API,   would you think  it's necessary to introduce a `Set<String> identifierColumns()` in `Schema` ?   I find the schema will always return the `fieldId` while the `UpdateSchema` will always use the `columnName` then we have to do the conversion from `fieldId` to `columnName` everywhere.

Reply:
@jackye1995, I don't think that the add/remove example you gave is idempotent because of the check `identifierFieldNames.contains(name)` that will result in an error: "Cannot remove column from identifier fields account_id because it is not an identifier field". @openinx gives a good example of why it's awkward to make users fall back to `remove`. And it is easy to make `set` work internally because you just clear the field ID set before calling `add` in a loop.

Reply:
Yes it is not idempotent, and using set to write add/remove is easier than using add/remove to write set, I will change to set then, thanks for the feedbacks.